File: OSCommon/org/optimizationservices/oscommon/datastructure/osentity/FunctionType.cs
using System;

namespace org.optimizationservices.oscommon.datastructure.osentity{
	/// <summary>
	/// A data structure class that corresponds to an xml element in the OSeL schema. 
	/// @author Jun Ma	
	/// @version 1.0, 09/01/2005
	/// @since OS 1.0
	/// @copyright (c) 2005
	/// </summary>
	public class FunctionType{
	
		public bool general = false;

		public bool basic = true;

		public bool arithmetic = true;

		public bool trigonometric = false;

		public bool statistics = false;

		public bool probability = false;

		public bool relationalAndLogic = false;

		public bool userFunction = false;

		public bool simulation = false;


	}//class FunctionType
}//namespace

File: OSCommon/src/org/optimizationservices/oscommon/datastructure/osentity/FunctionType.java
/**
 * @(#)FunctionType 1.0 03/14/2004
 *
 * Copyright (c) 2004
 */
package org.optimizationservices.oscommon.datastructure.osentity;


/**
 *
 * <p>A data structure class that corresponds to an xml element in the OSeL schema.  
 *  
 * </p>
 * @author Robert Fourer, Jun Ma, Kipp Martin
 * @version 1.0, 03/14/2004
 * @since OS 1.0
*/
public class FunctionType{
	
	public boolean general = false;

	public boolean basic = true;

	public boolean arithmetic = true;

	public boolean trigonometric = false;

	public boolean statistics = false;

	public boolean probability = false;

	public boolean relationalAndLogic = false;

	public boolean userFunction = false;

	public boolean simulation = false;

}//class FunctionType
